tools/bot/bot.go

```diff
@@ -0,0 +1,65 @@
+package bot
+
+import (
+	"errors"
+	"strings"
+)
+
+// BotInfo contains all information that is needed for a debugging message
+// TODO: This should later be extended with a custom message field that can be filled on the stream page
+type BotInfo struct {
+	CourseName   string
+	LectureHall  string
+	StreamUrl    string
+	CombIP       string
+	CameraIP     string
+	BotMessaging BotMessagingMethod
+}
+
+// BotMessagingMethod provides a generic interface for different message providers e.g. Matrix
+type BotMessagingMethod interface {
+	SendBotMessage(botInfo BotInfo) error
+}
+
+// SetMessagingMethod sets the provider method for sending messages e.g. matrix
+func (c *BotInfo) SetMessagingMethod(messaging BotMessagingMethod) {
+	c.BotMessaging = messaging
+}
+
+// getMessageText generates an unformatted issue text.
+func getMessageText(botInfo BotInfo) string {
+	return "🚨 Technical problem:" + "\n" +
+		"Course name: " + botInfo.CourseName + "\n" +
+		"Lecture hall: " + botInfo.LectureHall + "\n" +
+		"URL: " + botInfo.StreamUrl + "\n" +
+		"Lecture hall: " + botInfo.LectureHall + "\n" +
+		"CombinedIP: " + "http://" + botInfo.CombIP + "\n" +
+		"CameraIP: " + "http://" + botInfo.CameraIP + "\n"
```

Comment:
Is it possible to use http**s** instead?

Reply:
Unfortunately not :( This could be made configurable in the `LectureHall` at some point though